extension-style/src/main/java/com/mapbox/maps/extension/style/projection/generated/ProjectionExt.kt

```diff
@@ -0,0 +1,40 @@
+// This file is generated.
+@file:JvmName("StyleProjectionUtils")
+
+package com.mapbox.maps.extension.style.projection.generated
+
+import com.mapbox.maps.StylePropertyValueKind
+import com.mapbox.maps.extension.style.StyleInterface
+import com.mapbox.maps.extension.style.layers.properties.generated.ProjectionName
+import com.mapbox.maps.extension.style.utils.unwrapToAny
+
+/**
+ * Extension function to get the projection provided by the Style Extension.
+ *
+ * @return projection [Projection] that is currently applied to the map
+ */
+fun StyleInterface.getProjection(): Projection {
+  getStyleProjectionProperty("name").apply {
+    return Projection(
+      if (kind == StylePropertyValueKind.UNDEFINED) {
+        ProjectionName.MERCATOR
+      } else {
+        val projectionAsString = value.unwrapToAny() as String
+        ProjectionName.valueOf(projectionAsString.uppercase())
+      }
```

Comment:
Would it make more sense to check for `kind == string` and have the else be the branch that returns mercator as a fallback? Not sure if GL Native would ever return a property value with a different kind, but verifying string before we cast to string seems more robust.